### PR TITLE
[SPARK-36226][PYTHON][DOCS] Improve python docstring links to other classes

### PR DESCRIPTION
**Pull request by dominikgehl**
### What changes were proposed in this pull request?
additional links to other classes in python documentation

### Why are the changes needed?
python docstring syntax wasn't fully used everywhere

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Documentation change only

### CODE REVIEW

File: python/pyspark/sql/group.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ def pivot(self, pivot_col, values=None):
         ----------
         pivot_col : str
             Name of the column to pivot.
-        values :
+        values : list, optional
```
Comment (srowen) [inline python/pyspark/sql/group.py:241]: Do we use this convention anywhere else?

File: python/pyspark/sql/session.py [hunk 2]
```diff
@@ -73,7 +73,7 @@ class SparkSession(SparkConversionMixin):
 
     A SparkSession can be used create :class:`DataFrame`, register :class:`DataFrame` as
     tables, execute SQL over tables, cache tables, and read parquet files.
-    To create a SparkSession, use the following builder pattern:
+    To create a :class:`SparkSession`, use the following builder pattern:
```
Comment (srowen) [inline python/pyspark/sql/session.py:76]: This is fine, but do we do it consistently? if that would be a big change, I wonder how much it's worth it

File: python/pyspark/sql/group.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/group.py:241]: Yeah, it's from https://numpydoc.readthedocs.io/en/latest/format.html. I think it's fine.

Comment (HyukjinKwon) [reply python/pyspark/sql/group.py:241]: I would do it in batch though.

File: python/pyspark/sql/session.py (hunk 2 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/session.py:76]: Yeah, I agree.

Comment (dominikgehl) [reply python/pyspark/sql/session.py:76]: It seems to be already done in most classes below sql .. these were just a couple of exceptions I noticed

Comment (HyukjinKwon) [reply python/pyspark/sql/session.py:76]: If these are all of them, I think it's good to go.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master, branch-3.2 and branch-3.1.

**Pull request closed without merging**